packages/bot-web-ui/src/stores/summary-card-store.js

```diff
@@ -19,26 +32,59 @@ export default class SummaryCardStore {
     @computed
     get is_contract_completed() {
         return (
-            !!this.contract &&
-            !!this.contract.is_sold &&
+            !!this.contract_info &&
+            !!this.contract_info.is_sold &&
             this.root_store.run_panel.contract_stage !== contract_stages.PURCHASE_RECEIVED
         );
     }
 
     @computed
     get is_contract_loading() {
         return (
-            (this.root_store.run_panel.is_running && this.contract === null) ||
+            (this.root_store.run_panel.is_running && this.contract_info === null) ||
             this.root_store.run_panel.contract_stage === contract_stages.PURCHASE_SENT ||
             this.root_store.run_panel.contract_stage === contract_stages.STARTING
         );
     }
 
     @computed
     get is_contract_inactive() {
-        return !this.contract && !this.is_loading;
+        return !this.contract_info && !this.is_loading;
+    }
+
+    @action.bound
+    clear(should_unset_contract = true) {
+        if (should_unset_contract) {
+            this.contract_info = null;
+        }
+
+        this.profit = 0;
+        this.profit_loss = 0;
+        this.indicative = 0;
+        this.indicative_movement = '';
+        this.profit_movement = '';
+    }
+
+    @action.bound
+    clearContractUpdateConfigValues() {
+        if (this.contract_info) {
+            Object.assign(this, getContractUpdateConfig(this.contract_info));
+        }
     }
 
+    @action.bound
+    getLimitOrder = () => {
```

Comment:
I think there is no need to use arrow function here, since the `@action.bound` will bind the function itself.